### PR TITLE
fix(api): permission

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report it#11401.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report it#11426.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report cds-package#11445.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

File: engine/api/group/group.go [hunk 1]
```diff
@@ -35,12 +35,12 @@ func DeleteUserFromGroup(db gorp.SqlExecutor, groupID, userID int64) error {
 }
 
 // CheckUserInDefaultGroup insert user in default group
-func CheckUserInDefaultGroup(ctx context.Context, db gorp.SqlExecutor, userID int64) error {
-	if DefaultGroup == nil || DefaultGroup.ID == 0 {
+func CheckUserInDefaultGroup(ctx context.Context, db gorp.SqlExecutor, u *sdk.AuthentifiedUser) error {
```
Comment (fsamin) [inline engine/api/group/group.go]: I think the signature of this function was good

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#11464.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11464.0 ✘  
* Stage 1
  * Engine Test with DB ✘

File: ui/src/app/shared/keys/list/keys.list.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
                 <th class="two wide">{{ 'keys_name' | translate }}</th>
                 <th class="two wide">{{ 'keys_type' | translate }}</th>
                 <th class="ten wide">{{ 'keys_public' | translate }}</th>
-                <th class="two wide" *ngIf="edit === permission.READ_WRITE_EXECUTE"></th>
+                <th class="two wide" *ngIf="edit === true"></th>
```
Comment (sguiheux) [inline ui/src/app/shared/keys/list/keys.list.html]: if edit is boolean,  no need === true

File: ui/src/app/shared/keys/list/keys.list.html [hunk 3]
```diff
@@ -22,7 +22,7 @@
                 <td>
                     <pre>{{ k.public }}</pre>
                 </td>
-                <td *ngIf="edit === permission.READ_WRITE_EXECUTE">
+                <td *ngIf="edit === true">
```
Comment (sguiheux) [inline ui/src/app/shared/keys/list/keys.list.html]: if edit is boolean,  no need === true

File: ui/src/app/views/application/show/application.html [hunk 4]
```diff
@@ -69,14 +68,14 @@
                         </div>
                         <div *ngSwitchCase="'variables'">
                             <ng-container
-                                *ngIf="application.permission === perm.READ_WRITE_EXECUTE && !application.from_repository">
+                                *ngIf="project.permissions.writable === true && !application.from_repository">
```
Comment (sguiheux) [inline ui/src/app/views/application/show/application.html]: if edit is boolean,  no need === true

File: ui/src/app/views/application/show/application.html [hunk 5]
```diff
@@ -69,14 +68,14 @@
...
                                 <h3>{{ 'application_variable_form_title' | translate}}</h3>
                                 <app-variable-form [loading]="varFormLoading"
                                     (createVariableEvent)="variableEvent($event)"></app-variable-form>
                             </ng-container>
                             <h3>{{ 'application_variable_list_title' | translate}}</h3>
                             <app-variable [variables]="application.variables"
-                                [mode]="(application.permission === perm.READ_WRITE_EXECUTE && !application.from_repository)?'edit':'ro'"
+                                [mode]="(project.permissions.writable === true && !application.from_repository)?'edit':'ro'"
```
Comment (sguiheux) [inline ui/src/app/views/application/show/application.html]: if edit is boolean,  no need === true

File: ui/src/app/views/application/show/application.html [hunk 6]
```diff
@@ -86,7 +85,7 @@ <h3>{{ 'application_variable_list_title' | translate}}</h3>
                         <div *ngSwitchCase="'vuln'">
                             <app-vulnerabilities [project]="project" [application]="application"
                                 [vulnerabilities]="application.vulnerabilities"
-                                [edit]="(application.permission === perm.READ_WRITE_EXECUTE && !application.from_repository)">
+                                [edit]="project.permissions.writable === true">
```
Comment (sguiheux) [inline ui/src/app/views/application/show/application.html]: if edit is boolean,  no need === true

File: ui/src/app/views/workflow/show/workflow.html [hunk 7]
```diff
@@ -60,7 +60,7 @@
                                                     </i>
                                                 </div>
                                                 <div *ngIf="previewWorkflow" class="ui green button" (click)="savePreview()"
-                                                    [class.disabled]="loading || (detailedWorkflow.from_repository && detailedWorkflow.from_repository.length > 0) || detailedWorkflow.permission < permissionEnum.READ_WRITE_EXECUTE"
+                                                    [class.disabled]="loading || (detailedWorkflow.from_repository && detailedWorkflow.from_repository.length > 0) || detailedWorkflow.permission.writable === false"
```
Comment (sguiheux) [inline ui/src/app/views/workflow/show/workflow.html]: if edit is boolean,  no need === false

**Comment by ovh-cds** [comment]
CDS Report cds-package#11468.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

**Comment by ovh-cds** [comment]
CDS Report cds-package#11475.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

File: ui/src/app/shared/audit/list/audit.list.component.ts [hunk 8]
```diff
@@ -37,6 +38,7 @@ export class AuditListComponent {
             <Column<AuditWorkflow>>{
                 type: ColumnType.CONFIRM_BUTTON,
                 name: '',
+                disabled: !this.canRollback,
```
Comment (bnjjj) [inline ui/src/app/shared/audit/list/audit.list.component.ts]: it won't work. Must be in a ngOnInit because it's a variable coming from an @Input

**Comment by ovh-cds** [comment]
CDS Report cds-package#11482.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

**Pull request merged**